### ISSUE TITLE
Research Jquery uploader and status bar

### ISSUE THREAD

**Issue by crowesn** [uclibs/scholar_uc#1635]
### Descriptive summary

For large files, the status bar sits without providing meaningful feedback while processing in the background. Examine settings for this behavior.

**Comment by jayanthdeejay** [comment]
After setting up a local production environment to see if virus scanning is the culprit, I realized it is not the case. When a file is being uploaded, it is stored in the Virtual Memory[most possibly]/Main Memory. Once it reaches 100%, it does some processing (not sure what!) on the file. From the point of 0% upload until 100% upload and during the long wait after that till it is completely uploaded, application created an empty file in /tmp. Once the file processing is finished, it moves the entire file to /tmp. And after the given work is saved, the entire file is moved to Fedora.

This is an issue that exists in Hyrax Gem too, which I tested. After digging further, found that Hyrax is using a JQuery File Uploader in [Hyrax](https://github.com/samvera/hyrax/blob/v1.0.4/vendor/assets/javascripts/fileupload/jquery.fileupload.js) provided by [blueimp](https://github.com/blueimp/jQuery-File-Upload/blob/master/js/jquery.fileupload.js). Setting the right bugger size or disabling the multipart upload might resolve this issue which I need to test. Other applications that use JQuery file uploader has the same issue which could be found https://github.com/rack/rack/issues/1075.

**Comment by jayanthdeejay** [comment]
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/samvera-tech/nyBfqT1kWXM/ZPP6i-CxAAAJ

**Comment by jayanthdeejay** [comment]
Apparently, JQuery file uploader uses https://github.com/rack/rack/ gem to handle caching and parsing files. Issue is in Rack gem and it was fixed recently. Check https://github.com/rack/rack/pull/1192.

I tested it by manually increasing the buffer size to 1MB in my local rack gem and it did finish uploading 524MB file in less than 30 secs. Updating rack gem to the mentioned commit resolves the issue.
